scripts/dmodels_spawning.dsc

```diff
@@ -83,13 +156,59 @@ dmodels_delete:
     - remove <[root_entity].flag[dmodel_parts]>
     - remove <[root_entity]>
 
-dmodels_reset_model_position:
+dmodels_set_yaw:
+    type: task
+    debug: false
+    definitions: root_entity[EntityTag]|yaw[Number]|update[True/False]
+    description: Sets the yaw of the model
+    script:
+    - flag <[root_entity]> dmodel_yaw:<[yaw]>
+    - if <[update]||true>:
+        - run dmodels_reset_model_position def.root_entity:<[root_entity]>
+
+dmodels_set_rotation:
+    type: task
+    debug: false
+    definitions: root_entity[EntityTag]|quaternion[QuaternionTag]|update[True/False]
+    description: Sets the global rotation of a model
+    script:
+    - if <quaternion[<[quaternion]>]||null> == null:
+        - debug error "<&[error]>Invalid input the rotation must be a quaternion."
+    - flag <[root_entity]> dmodel_global_rotation:<[quaternion]>
+    - if <[update]||true>:
+        - run dmodels_reset_model_position def.root_entity:<[root_entity]>
+
+dmodels_set_scale:
+    type: task
+    debug: false
+    definitions: root_entity[EntityTag]|scale[LocationTag]|update[True/False]
+    description: Sets the global scale of the model
+    script:
+    - if <location[<[scale]>]||null> == null:
+        - debug error "<&[error]>Invalid input the scale must be a location."
+    - flag <[root_entity]> dmodel_global_scale:<[scale]>
+    - if <[update]||true>:
+        - run dmodels_reset_model_position def.root_entity:<[root_entity]>
+
+dmodels_set_color:
+    type: task
+    debug: false
+    definitions: root_entity[EntityTag]|color[ColorTag]
+    script:
+    - if <color[<[color]>]||null> == null:
+        - debug error "<&[error]>Invalid input must be a color."
+    - flag <[root_entity]> dmodels_color:<[color]>
+    - foreach <[root_entity].flag[dmodel_parts]> as:part:
+        - define item <[part].item>
+        - adjust <[item]> color:<[color]> save:item
+        - define new_item <entry[item].result>
+        - adjust <[part]> item:<[new_item]>
+
+dmodels_set_view_range:
     type: task
     debug: false
-    definitions: root_entity
+    definitions: root_entity[EntityTag]|view_range[Number]
     script:
-    - define center <[root_entity].location.with_pitch[0].below[0.72]>
-    - define yaw_mod <[root_entity].location.yaw.add[180].to_radians>
+    - flag <[root_entity]> dmodel_view_range:<[view_range]>
     - foreach <[root_entity].flag[dmodel_parts]> as:part:
-        - adjust <[part]> armor_pose:[head=<[part].flag[dmodel_def_pose]>]
-        - teleport <[part]> <[center].add[<[part].flag[dmodel_def_offset].rotate_around_y[<[yaw_mod].mul[-1]>]>]>
+        - adjust <[part]> view_range:<[view_range]>
```

Comment:
a `foreach` isn't needed as the input to a single `adjust` command, just put the list in directly